src/views/Project/index.jsx

```diff
@@ -238,12 +271,42 @@ export default class Project extends Component {
     });
   };
 
-  load = async () => {
+  load = () => {
+    this.setState({ isNextPageLoading: true });
+
     const project = projects[this.props.match.params.project];
     const repositories = mergeAll(project.repositories);
     const tagsMapping = tagReposMapping(repositories);
+    const productWithComponentList = mergeAll(
+      project.products
+        ? project.products.filter(product => typeof product !== 'string')
+        : []
+    );
+    const productWithNoComponentList = project.products
+      ? project.products.filter(product => typeof product === 'string')
+      : [];
 
-    await Promise.all(
+    if (
+      !(
+        Object.entries(productWithComponentList).some(
```

Comment:
nits. if you only need the key of the mapping, you can use `Object.keys`
```suggestion
        Object.keys(productWithComponentList).some(products =>
```